src/main/java/appeng/core/sync/packets/PacketAssemblerAnimation.java

```diff
@@ -22,65 +22,96 @@
 import java.io.IOException;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
-import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.world.World;
 
-import cpw.mods.fml.relauncher.Side;
-import cpw.mods.fml.relauncher.SideOnly;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
 
 import appeng.api.storage.data.IAEItemStack;
+import appeng.client.ClientHelper;
 import appeng.client.EffectType;
 import appeng.core.CommonHelper;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.util.item.AEItemStack;
 
 
-public class PacketAssemblerAnimation extends AppEngPacket
+public class PacketAssemblerAnimation implements AppEngPacket, AppEngPacketHandler<PacketAssemblerAnimation, AppEngPacket>
 {
 
-	public final int x;
-	public final int y;
-	public final int z;
-	public final byte rate;
-	public final IAEItemStack is;
+	private int x;
+	private int y;
+	private int z;
+	private byte rate;
+	private IAEItemStack is;
 
 	// automatic.
-	public PacketAssemblerAnimation( ByteBuf stream ) throws IOException
+	public PacketAssemblerAnimation()
 	{
-		this.x = stream.readInt();
-		this.y = stream.readInt();
-		this.z = stream.readInt();
-		this.rate = stream.readByte();
-		this.is = AEItemStack.loadItemStackFromPacket( stream );
 	}
 
 	// api
-	public PacketAssemblerAnimation( int x, int y, int z, byte rate, IAEItemStack is ) throws IOException
+	public PacketAssemblerAnimation( int x, int y, int z, byte rate, IAEItemStack is )
 	{
+		this.x = x;
+		this.y = y;
+		this.z = z;
+		this.rate = rate;
+		this.is = is;
+	}
 
-		ByteBuf data = Unpooled.buffer();
+	@Override
+	public AppEngPacket onMessage( PacketAssemblerAnimation message, MessageContext ctx )
+	{
+		final double d0 = 0.5d;// + ((double) (Platform.getRandomFloat() - 0.5F) * 0.26D);
+		final double d1 = 0.5d;// + ((double) (Platform.getRandomFloat() - 0.5F) * 0.26D);
+		final double d2 = 0.5d;// + ((double) (Platform.getRandomFloat() - 0.5F) * 0.26D);
+		final World world = ClientHelper.proxy.getWorld();
 
-		data.writeInt( this.getPacketID() );
-		data.writeInt( this.x = x );
-		data.writeInt( this.y = y );
-		data.writeInt( this.z = z );
-		data.writeByte( this.rate = rate );
-		is.writeToPacket( data );
-		this.is = is;
+		CommonHelper.proxy.spawnEffect( EffectType.Assembler, world, message.x + d0, message.y + d1, message.z + d2, message );
+
+		return null;
+	}
 
-		this.configureWrite( data );
+	@Override
+	public void fromBytes( ByteBuf buf )
+	{
+		this.x = buf.readInt();
+		this.y = buf.readInt();
+		this.z = buf.readInt();
+		this.rate = buf.readByte();
+		try
+		{
+			this.is = AEItemStack.loadItemStackFromPacket( buf );
+		}
+		catch( IOException e )
+		{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)